### PR TITLE
Refactor MetaBoxesArea to to functional components using hooks.

### PR DESCRIPTION
**Pull request by torounit**
## Description

Related: #22890 his PR refactors the `MetaBoxesArea` component into a function component using React hooks.

## How has this been tested?

Show block editor and check custom field metabox and others.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] I've tested my changes with keyboard and screen readers. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

File: packages/edit-post/src/components/meta-boxes/meta-boxes-area/index.js [hunk 1]
```diff
@@ -6,83 +6,54 @@ import classnames from 'classnames';
...
-	return {
-		isSaving: select( editPostStore ).isSavingMetaBoxes(),
-	};
-} )( MetaBoxesArea );
+		};
+	}, [ form ] );
+
+	const isSaving = useSelect( ( select ) => {
+		return select( editPostStore ).isSavingMetaBoxes();
+	} );
```
Comment (Mamaduka) [inline packages/edit-post/src/components/meta-boxes/meta-boxes-area/index.js]: Let's optimize this `useSelect` call by adding a dependency array.

File: packages/edit-post/src/components/meta-boxes/meta-boxes-area/index.js [hunk 2]
```diff
@@ -6,83 +6,54 @@ import classnames from 'classnames';
...
-	}
-}
-
-export default withSelect( ( select ) => {
-	return {
-		isSaving: select( editPostStore ).isSavingMetaBoxes(),
-	};
-} )( MetaBoxesArea );
+		};
+	}, [ form ] );
```
Comment (Mamaduka) [inline packages/edit-post/src/components/meta-boxes/meta-boxes-area/index.js]: Maybe we should include `location` here as a dependency.

**Comment by Mamaduka** [comment]
Hi, @torounit

Thanks for working on this.

~~Waiting for the build process to finish and will test it locally.~~ Tested locally with `Gutenberg Test Plugin, Meta Box` and works as expected. Besides the two comments I left, the conversion looks good to me.

File: packages/edit-post/src/components/meta-boxes/meta-boxes-area/index.js (hunk 2 above)
Comment (torounit) [reply packages/edit-post/src/components/meta-boxes/meta-boxes-area/index.js]: Thanks catch!

**Comment by Mamaduka** [comment]
@torounit, sorry for the late follow-up on this PR.

Can you rebase the current trunk? I'll do another round of tests, and we're good to land this 👍

**Comment by torounit** [comment]
@Mamaduka 

I made a few changes. And I've included trunk. Can I have you test it?

**Comment by Mamaduka** [comment]
Thanks, @torounit

I will test this first thing in the morning tomorrow 👍

**Review by Mamaduka — APPROVED** [review]
LGTM 👍

**Comment by Mamaduka** [comment]
@torounit

Sidenote: I think it's preferred method to use `function Name () {}` for components. See: https://github.com/WordPress/gutenberg/issues/22890#issuecomment-638949807.

But this isn't a blocker.

**Comment by Mamaduka** [comment]
@torounit, I think we're good to merge this PR.

**Pull request merged**